thing/index.js

```diff
@@ -514,9 +549,20 @@ function ThingShadowsClient(deviceOptions, thingShadowOptions) {
          // which the application will need to filter out.
          //
          if (thingShadows[thingName].persistentSubscribe === true) {
-            this._handleSubscriptions(thingName, ['update', 'get', 'delete'], ['accepted', 'rejected'],
-               'subscribe');
+            topicsObject.push({
+               operations: ['update', 'get', 'delete'],
+               statii: ['accepted', 'rejected'],
+            });
+         }
+
+         if (topicsObject.length > 0) {
+            this._handleSubscriptions(thingName, topicsObject, 'subscribe', callback);
```

Comment:
From encapsulation point of view, pending should probably better be manipulated here, also you probably don't want to return granted etc in this register api, as whoever calls register() most likely won't care about what topics are accepted or not, what they do care is whether or not register() is successful. 

I would recommend to do something like this here

```
this._handleSubscriptions(thingName, topicsObject, 'subscribe', function(granted) {
        var res = "Not all subscriptions were granted";
        if (all topics are granted) {
               res = null;
               this.pending = false;  
        }
        if (!undefined(callback)) {
             callback(res);
        }
});
```


Also note _handleSubscriptions() does subscribe and unsubscribe, but for unsubscribe, there's no granted argument to the callback function.